### PR TITLE
FIX: make other checkboxes read-only while updating the one.

### PR DESCRIPTION
**Pull request by vinothkannans**
Else checking multiple checkboxes discards edits to all but the first box. This is happening when clicking checkboxes in a small time of interval.

### CODE REVIEW

File: assets/javascripts/discourse/initializers/checklist.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ export function checklistSyntax($elem, postDecorator) {
   const $boxes = $elem.find(".chcklst-box");
   const postWidget = postDecorator.widget;
   const postModel = postDecorator.getModel();
+  const removeReadonlyClass = () => $boxes.removeClass("readonly");
```
Comment (jjaffeux) [inline assets/javascripts/discourse/initializers/checklist.js:22]: We shouldn't add any new jquery code, either remove all jquery of this file if you have the time, or make this specific call using vanilla js

File: assets/javascripts/discourse/initializers/checklist.js [hunk 2]
```diff
@@ -28,16 +29,18 @@ export function checklistSyntax($elem, postDecorator) {
     $(val).click((ev) => {
       const $box = $(ev.currentTarget);
 
-      if ($box.hasClass("permanent")) {
+      if ($box.hasClass("permanent") || $box.hasClass("readonly")) {
```
Comment (jjaffeux) [inline assets/javascripts/discourse/initializers/checklist.js:32]: same note on jquery

File: assets/javascripts/discourse/initializers/checklist.js [hunk 3]
```diff
@@ -28,16 +29,18 @@ export function checklistSyntax($elem, postDecorator) {
...
         return;
       }
 
+      $boxes.addClass("readonly");
```
Comment (jjaffeux) [inline assets/javascripts/discourse/initializers/checklist.js:36]: jquery

File: assets/javascripts/discourse/initializers/checklist.js [hunk 4]
```diff
@@ -100,13 +103,17 @@ export function checklistSyntax($elem, postDecorator) {
           });
 
           if (save && save.then) {
-            save.then(() => {
-              postWidget.attrs.isSaving = false;
-              postWidget.scheduleRerender();
-            });
+            save
+              .then(() => {
+                postWidget.attrs.isSaving = false;
```
Comment (jjaffeux) [inline assets/javascripts/discourse/initializers/checklist.js:108]: I no you didn't introduce this, but not a fan of this pattern where we mutate input data

File: assets/javascripts/discourse/initializers/checklist.js (hunk 1 above)
Comment (jjaffeux) [reply assets/javascripts/discourse/initializers/checklist.js:22]: I'm also concerned this anonymous function is holding reference to DOM which will make it leak most likely

File: assets/javascripts/discourse/initializers/checklist.js [hunk 5]
```diff
@@ -100,13 +115,17 @@ export function checklistSyntax($elem, postDecorator) {
           });
 
           if (save && save.then) {
-            save.then(() => {
-              postWidget.attrs.isSaving = false;
-              postWidget.scheduleRerender();
-            });
+            save
+              .then(() => {
+                postWidget.attrs.set("isSaving", false);
```
Comment (vinothkannans) [inline assets/javascripts/discourse/initializers/checklist.js:120]: @jjaffeux sorry, I'm unable to find any other way to do this. https://github.com/discourse/discourse-checklist/pull/57#discussion_r887668107

File: test/javascripts/lib/checklist-test.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ async function prepare(raw) {
   const decoratorHelper = { getModel: () => model };
 
   const $elem = $(cooked.string);
-  checklistSyntax($elem, decoratorHelper);
+  checklistSyntax($elem[0], decoratorHelper);
```
Comment (CvX) [inline test/javascripts/lib/checklist-test.js:18]: This broke the tests 🙂 you were passing only whatever the first element happened to be 😛 


This works:

```suggestion
  const $elem = $(`<div>${cooked.string}</div>`);
  checklistSyntax($elem[0], decoratorHelper);
```

**Review by CvX — APPROVED** [review]

**Pull request merged**